### PR TITLE
Improve the NACK message processing on istio agent.

### PR DESCRIPTION
**Pull request by myidpt**
Fixes https://github.com/istio/istio/issues/21111.

This PR follows https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol#nack.
Envoy returns NACK only when it finds the key/cert are invalid.
So Istio agent will treat NACK as ACK (holding), log the error. This is the same as Pilot. Istio agent will push new secret when there's an update.
For the gateway SDS, the admin inspects the log to find the error and push a new key/cert to trigger the secret reload.
For sidecar SDS, if this happens, it means there's a bug in CA / istio agent, which is not recoverable through retries anyway.

Also improved the following code:
1. Use cleaner version number.
2. Skip checking the node ID after the first request in the same connection.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ X ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by myidpt** [comment]
Adding more unit tests.

**Comment by myidpt** [comment]
I'm not super clear about the Envoy implementation. If there's no temp error on Envoy side, this sounds reasonable and we can drop the connection immediately.

**Comment by howardjohn** [comment]
if you just drop the connection you will have the same issue of an infinite
loop of `connect - send bad response - get nack - disconnect - reconnect` I
think

**Comment by myidpt** [comment]
How does Pilot handle NACK for xDS?

**Comment by myidpt** [comment]
Discussed with John and Jimmy. Envoy returns NACK only when it finds the key/cert are invalid.
So Istio agent will treat NACK as ACK (holding), log the error. This is the same as Pilot. Istio agent will push new secret when there's an update.
For the gateway SDS, the admin inspects the log to find the error and push a new key/cert to trigger the secret reload.
For sidecar SDS, if this happens, it means there's a bug in CA / istio agent, which is not recoverable through retries.

File: security/pkg/nodeagent/cache/secretcache.go [hunk 1]
```diff
@@ -803,7 +803,7 @@ func (sc *SecretCache) generateSecret(ctx context.Context, token string, connKey
 		Token:            token,
 		CreatedTime:      t,
 		ExpireTime:       expireTime,
-		Version:          t.String(),
+		Version:          t.Format("01-02 15:04:05.000"), // Precise enough version based on creation time.
```
Comment (howardjohn) [inline security/pkg/nodeagent/cache/secretcache.go]: why this change?
